rdt/hyper_transformer.py

```diff
@@ -16,34 +16,73 @@ class HyperTransformer(object):
     """Multitable transformer.
 
     Arguments:
-        meta_file(str): Path to the meta.json file
-        table_dict(dict): Dict with the table structure (table_name -> (data, meta))
-
-    Returns:
-        HyperTransformer
+        metadata(str or dict): Path to the meta.json file or its parsed contents.
 
     The main propouse of the HyperTransformer class is to easily manage transformations
     for a whole dataset.
-
-    It's required that one of `table_dict` or `meta_file` to be not null. In case both were
-    present `meta_file` takes precedence over `table_dict`.
     """
 
-    def __init__(self, meta_file=None, table_dict=None):
+    def __init__(self, metadata):
 
         self.transformers = {}  # key=(table_name, col_name) val=transformer
+        tables = None
```

Comment:
The usage of this variable is confusing, and it seems wrong.
Note that with the current implementation there are two different scenarios:

* Case 1: `metadata` is a `dict`:
    * `tables` is kept as `None`
    * `self.table_dict` is equal to `metadata`
    * `metadata` is passed to `self.get_transformers`
* Case 2: `metadata` is a `str`:
    * `tables` contains the output of the `self.get_tables call`
    * `metadata` contains the contents of the json file
    * `self.table_dict` is equal to `tables`, which is different than `metadata`
    * `metadata` is passed to `self.get_transformers`

Note how in the first scenario, `self.get_transformers` is passed the contents of `self.table_dict`, while in the second one it is passed a different thing.

Also, it is not clear to me why you should be passing `metadata` to `self.get_transformers` if `metadata` is already stored as the `self.tables_dict` attribute.

Lastly, if `self.get_tables` and `self.get_transformers` are only for internal usage, they should be "private" (so, start with an underscore)